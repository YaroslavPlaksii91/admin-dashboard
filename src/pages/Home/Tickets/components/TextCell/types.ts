export type TextCellType = {
  text: string;
  subtext: string;
};
