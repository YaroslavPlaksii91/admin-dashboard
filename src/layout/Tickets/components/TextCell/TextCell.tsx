import { FC } from 'react';
import { Typography, Box } from '@mui/material';

import { TextCellType } from './types';

export const TextCell: FC<TextCellType> = ({ text, subtext }) => {
  return (
    <Box>
      <Typography
        sx={{
          mb: 1,
          fontSize: 14,
          fontWeight: 600,
          lineHeight: 1.43,
          letterSpacing: 0.2,
        }}
      >
        {text}
      </Typography>
      <Typography
        sx={{
          fontSize: 12,
          lineHeight: 1.33,
          letterSpacing: 0.1,
          color: 'additionalInfoColor',
        }}
      >
        {subtext}
      </Typography>
    </Box>
  );
};
