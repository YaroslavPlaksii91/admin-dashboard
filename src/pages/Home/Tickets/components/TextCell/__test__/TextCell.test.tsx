import { cleanup, render, screen } from '@testing-library/react';

import { TextCell } from '../TextCell';

describe('TextCell component', () => {
  afterEach(cleanup);

  const text = 'Main Text';
  const subtext = 'Sub Text';

  it('should be rendered', () => {
    const { asFragment } = render(<TextCell text={text} subtext={subtext} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders text and subtext correctly', () => {
    render(<TextCell text={text} subtext={subtext} />);

    const mainTextElement = screen.getByText(text);
    const subTextElement = screen.getByText(subtext);

    expect(mainTextElement).toBeInTheDocument();
    expect(subTextElement).toBeInTheDocument();
  });

  it('applies styles correctly', () => {
    render(<TextCell text={text} subtext={subtext} />);

    const mainTextElement = screen.getByText(text);
    const subTextElement = screen.getByText(subtext);

    expect(mainTextElement).toHaveStyle('font-size: 14px');
    expect(mainTextElement).toHaveStyle('font-weight: 600');
    expect(subTextElement).toHaveStyle('color: additionalInfoColor');
  });
});
