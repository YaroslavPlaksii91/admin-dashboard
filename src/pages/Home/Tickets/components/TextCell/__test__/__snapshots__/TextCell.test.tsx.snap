// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextCell component should be rendered 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-k62y85-MuiTypography-root"
    >
      Main Text
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1edc1f9-MuiTypography-root"
    >
      Sub Text
    </p>
  </div>
</DocumentFragment>
`;
